src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java

```diff
@@ -195,49 +193,51 @@ private ParserResult loadDatabase(Path file) throws Exception {
 
         preferencesService.getFilePreferences().setWorkingDirectory(fileToLoad.getParent());
 
-        ParserResult result = null;
+        ParserResult parserResult = null;
         if (BackupManager.backupFileDiffers(fileToLoad)) {
-            result = BackupUIManager.showRestoreBackupDialog(dialogService, fileToLoad, preferencesService).orElse(null);
+            // In case the backup differs, ask the user what to do.
+            // In case the user opted for restoring a backup, the content of the backup is contained in parserResult.
+            parserResult = BackupUIManager.showRestoreBackupDialog(dialogService, fileToLoad, preferencesService).orElse(null);
         }
 
         try {
-            if (result == null) {
-                result = OpenDatabase.loadDatabase(fileToLoad,
+            if (parserResult == null) {
+                // No backup was restored, do the "normal" loading
```

Comment:
Well done for including the comments. I wrote that, but if I had to read it again in a year, I wouldn't know what is going on. I guess comments aren't useless after all.